### PR TITLE
Added Lamp Effect Component

### PR DESCRIPTION
**Pull request by rahul-vashishtha**

### CODE REVIEW

File: components/content/inspira/ui/lamp-effect/LampEffect.vue [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import { cn } from "~/lib/utils";
+
+interface LampEffectProps {
+  delay?: number;
+  duration?: number;
+  class?: HTMLAttributes["class"];
+}
+
+const props = withDefaults(defineProps<LampEffectProps>(), {
+  delay: 0.5,
```
Comment (nathan-de-pachtere) [inline components/content/inspira/ui/lamp-effect/LampEffect.vue:94]: Could be great to be consistent between components props across Inspira : if we are using `ms`, ` s` or other. What do you think ?

**Comment by rahul-vashishtha** [comment]
The transform: translateZ(1px) is the properties getting added to Spotlight and Glowing Light which should not be there.

Screenshot 2024-10-24 at 12 26 34 PM
Screenshot 2024-10-24 at 12 26 27 PM

**Comment by rahul-vashishtha** [comment]
Once this is resolved, the port will be ready for review.

File: components/content/inspira/ui/lamp-effect/LampEffect.vue [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      <!-- Spotlight -->
+      <div
+        v-motion
+        :initial="spotLightInitial"
+        :visible="spotLightAnimate"
+        class="absolute inset-auto z-30 h-36 w-64 -translate-y-[6rem] rounded-full bg-cyan-400 blur-2xl"
+      ></div>
+
+      <!-- Glowing Line -->
+      <div
```
Comment (nathan-de-pachtere) [inline components/content/inspira/ui/lamp-effect/LampEffect.vue:66]: <div
        ref="glowingLight"
        class="absolute inset-auto z-50 h-0.5 w-[30rem] -translate-y-[7rem] bg-cyan-400"
      ></div>

File: components/content/inspira/ui/lamp-effect/LampEffect.vue [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+const spotLightAnimate: Variant = {
+  width: "16rem",
+  transition: {
+    delay: props.delay * 1000,
+    duration: props.duration * 1000,
+    ease: "easeInOut",
+  },
+};
+
+const glowingLineInitial: Variant = { width: "15rem" };
```
Comment (nathan-de-pachtere) [inline components/content/inspira/ui/lamp-effect/LampEffect.vue:120]:
```typescript
import gsap from 'gsap';

const glowingLight = ref<HTMLElement>();

onMounted(() => {
  gsap.to(glowingLight, {width: '30rem' , duration: props.duration * 1000, delay: props.delay * 1000, ease: "easeInOut"})
})
```

**Comment by nathan-de-pachtere** [comment]
@rahul-vashishtha Your problem it's a side effect of vueuse/motion as you mention, and I think for motion stuff it's better to use framework-agnostic library. Not only for motion but for every stuff that are browser native. Vueuse/motion is not so much active, so not good to include.

The code changes above are just an example with GSAP, but you could do the same with animate.js.

File: components/content/inspira/ui/lamp-effect/LampEffect.vue (hunk 1 above)
Comment (rahul-vashishtha) [reply components/content/inspira/ui/lamp-effect/LampEffect.vue:82]: Changed to using seconds.

File: components/content/inspira/ui/lamp-effect/LampEffect.vue (hunk 3 above)
Comment (rahul-vashishtha) [reply components/content/inspira/ui/lamp-effect/LampEffect.vue]: Using only CSS keyframes for now.

**Comment by rahul-vashishtha** [comment]
@nathan-de-pachtere I have removed the `vueuse/motion` and made the animation css only.

I'm thinking of removing `vueuse/motion` permanently from the project in near future as it creates a lot of issues with different components.

**Comment by nathan-de-pachtere** [comment]
Super nice in CSS only ;)

**Pull request merged**